WordPress/Classes/ViewRelated/Reader/Controllers/ReaderSearchSuggestionsView.swift

```diff
@@ -0,0 +1,114 @@
+import UIKit
+import SwiftUI
+import WordPressShared
+
+struct ReaderSearchSuggestionsView: View {
+    @ObservedObject var viewModel: ReaderSearchSuggestionsViewModel
+
+    var body: some View {
+        List {
+            ForEach(viewModel.suggestions.prefix(7), id: \.self) { suggestion  in
+                Button {
+                    viewModel.onSelection?(suggestion.searchPhrase)
+                } label: {
+                    makeItem(for: suggestion)
+                }
+            }
+            .onDelete(perform: viewModel.delete)
+
+            if !viewModel.suggestions.isEmpty {
+                Button {
+                    viewModel.buttonClearSearchHistoryTapped()
+                } label: {
+                    Text(Strings.clearHistory)
+                        .foregroundStyle(AppColor.brand)
+                }
+            }
+        }
+        .listStyle(.plain)
+    }
+
+    private func makeItem(for suggestion: ReaderSearchSuggestion) -> some View {
+        HStack {
+            Text(suggestion.searchPhrase)
+            Spacer()
+            Button {
+                viewModel.delete([suggestion])
+            } label: {
+                Image(systemName: "xmark")
+                    .font(.footnote)
+                    .foregroundStyle(.secondary)
+            }
+            .buttonStyle(.plain)
+        }
+        .swipeActions(edge: .trailing) {
+            Button(SharedStrings.Button.delete, role: .destructive) {
+                viewModel.delete([suggestion])
+            }.tint(.red)
+        }
+    }
+}
+
+final class ReaderSearchSuggestionsViewModel: ObservableObject {
+    @Published private(set) var suggestions: [ReaderSearchSuggestion] = []
+    @Published private(set) var allSuggestions: [ReaderSearchSuggestion] = []
+
+    var onSelection: ((String) -> Void)?
+
+    private let coreData = ContextManager.shared
+
+    init() {
+        reloadSuggestions()
+    }
+
+    var searchText: String = "" {
+        didSet { updateDisplayedSuggesions() }
+    }
+
+    private func reloadSuggestions() {
+        let request = NSFetchRequest<ReaderSearchSuggestion>(entityName: "ReaderSearchSuggestion")
+        request.sortDescriptors = [NSSortDescriptor(key: "date", ascending: false)]
+        self.allSuggestions = (try? coreData.mainContext.fetch(request)) ?? []
+        self.updateDisplayedSuggesions()
+    }
+
+    private func updateDisplayedSuggesions() {
+        let searchText = searchText.trimmingCharacters(in: .whitespacesAndNewlines)
+        if searchText.isEmpty {
+            suggestions = allSuggestions
+        } else {
+            suggestions = StringRankedSearch(searchTerm: searchText)
+                .search(in: allSuggestions, input: \.searchPhrase)
+        }
+    }
+
+    func delete(at indexSet: IndexSet) {
+        delete(indexSet.map { suggestions[$0] })
+    }
+
+    func delete(_ suggestions: [ReaderSearchSuggestion]) {
+        let context = coreData.mainContext
```

Comment:
I'm aware that the original view controller implementation operates in the main context. But what do you think about using `ContextManager.performAndSave` to delete in the new implementation here?

Reply:
Or, maybe not using Core Data at all.

Reply:
> Or, maybe not using Core Data at all.

That was my original intention, but I decided to reduce the number of changes. I made some preparations by removing `NSFetchedResultsController`, so I was actually almost there. I ended up [removing](https://github.com/wordpress-mobile/WordPress-iOS/pull/23767/commits/34b82b4e4c44654b9d2e5e1fbe371925a1ac3c41) Core Data usage as you suggested.

Heads up: I forgot to merge and rebased on instead, so the history is updated.